modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/SyncContentServiceImpl.java

```diff
@@ -0,0 +1,150 @@
+package com.enonic.xp.core.impl.content;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+
+import com.enonic.xp.content.ContentService;
+import com.enonic.xp.content.ContentSynchronizer;
+import com.enonic.xp.content.ResetContentInheritParams;
+import com.enonic.xp.content.SyncContentService;
+import com.enonic.xp.content.processor.ContentProcessor;
+import com.enonic.xp.core.impl.content.processor.ContentProcessors;
+import com.enonic.xp.core.impl.content.serializer.ContentDataSerializer;
+import com.enonic.xp.event.EventPublisher;
+import com.enonic.xp.node.NodeService;
+import com.enonic.xp.page.PageDescriptorService;
+import com.enonic.xp.project.ProjectService;
+import com.enonic.xp.region.LayoutDescriptorService;
+import com.enonic.xp.region.PartDescriptorService;
+import com.enonic.xp.schema.content.ContentTypeService;
+
+@Component
+public class SyncContentServiceImpl
+    implements SyncContentService
+{
+    private final ContentProcessors contentProcessors = new ContentProcessors();
+
+    private ContentService contentService;
+
+    private ContentTypeService contentTypeService;
+
+    private NodeService nodeService;
+
+    private EventPublisher eventPublisher;
+
+    private ContentNodeTranslator translator;
+
+    private PageDescriptorService pageDescriptorService;
+
+    private PartDescriptorService partDescriptorService;
+
+    private LayoutDescriptorService layoutDescriptorService;
+
+    private ContentDataSerializer contentDataSerializer;
```

Comment:
Codacy found an issue: [Perhaps 'contentDataSerializer' could be replaced by a local variable.](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=6324450)